src/Cassandra.IntegrationTests/Core/PreparedStatementsTests.cs

```diff
@@ -858,6 +860,65 @@ private void AssertValid(ISession session, PreparedStatement ps, object value)
             }
         }
 
+        [Test]
+        [TestCassandraVersion(2, 0)]
+        public void Batch_PreparedStatement_With_Unprepared_Flow()
```

Comment:
Since most batch integration tests have been migrated to use simulacron, I was going to ask that we include some tests using a real ccm cluster, but you already thought of that :)

Reply:
actually this was more due to a limitation with simulacron that prevented me from moving these tests but yeah I agree with you that we should have some real cluster tests  here